src/api/v4/dataview/base.js

```diff
@@ -156,7 +159,9 @@ Base.prototype.removeFilter = function (filter) {
  */
 Base.prototype.hasFilter = function (filter) {
   this._checkFilter(filter);
-  return (filter instanceof BoundingBoxLeafletFilter) && (filter === this._boundingBoxFilter) && (this._internalModel && this._internalModel.get('sync_on_bbox_change'));
+  return (filter === this._boundingBoxFilter) &&
+         (this._internalModel && this._internalModel.get('sync_on_bbox_change')) &&
+         (filter instanceof BoundingBoxFilter || filter instanceof BoundingBoxLeafletFilter);
```

Comment:
Would we be able to simplify this if `BoundingBoxLeafletFilter` was a subclass of `BoundingBoxFilter` (just doing `filter instance of BoundingBoxFilter`?

Reply:
`BoundingBoxLeafletFilter` in a "brother" of `BoundingBox` not a child. Inheritance can't be done here because `BoundingBox` exposes "setBounds" method that is not contained in `BoundingBoxLeafletFilter`.

Reply:
But they both share a common interface (`getBounds` and `boundsChanged` event) which could be part of a parent class. `setBounds` is something external...

Reply:
We could add a BoundingBox.Base with this interface. Sounds good to you?